atlp-rwanda/phantom-be-elite
# #18 Chore: Setup Continuous Integration & Quality Assurance pipeline

#### What does this PR do?

- Configuring testing environment
- Configuring continous integration and deployement 
- Adds Github actions badge to the repo

#### Description of Task to be completed?

- Codes are continously tested before the deployment 
- Github action badge are displayed
- Codeclimate to ensure the code analysis and maintainability checking.

#### How should this be manually tested?

- Clone the repo

- checkout to branch `ch-setup-integrations-18`

- Run `npm run install` to install the dependencies

- Run `npm run coverage` to run testing coverage report

- Run `npm run test` to run tests

- Check the ReadMe of the repo to confirm the github actions badges are displayed

#### What are the relevant Pivotal Tracker/Trello stories?

- https://trello.com/c/WgAyuZBs/18-setup-continuous-integration-quality-assurance-pipeline

#### Screenshots (if appropriate)

-  N/A

#### Questions:

-  N/A
